### ISSUE TITLE
How to display all fields from Schema when provided json does not all the field

### ISSUE THREAD

**Issue by MahendraBalu** [json-editor/json-editor#208]
### Expected behavior
1. Schema have 2 fields with none of them are required.
2. JSON have 1 field
3. UI should display both the fields so that user can provide the value for the missing field in JSON.
### Actual behavior
UI is displaying only field provided in JSON.

**Comment by pmk65** [comment]
Probably the same problem as in #163

**Comment by MahendraBalu** [comment]
Do we have any work around ?? 
Is there any plans to fix this bug?

**Comment by schmunk42** [comment]
I think they should (have to) be required. Which just requires the existence of a key, but not a value.

**Comment by MahendraBalu** [comment]
It is optional field as per the schema. If include in required than user has to provide the value.

**Comment by schmunk42** [comment]
> If include in required than user has to provide the value

It should save an empty string.

Can you post an example or demo-link. 


https://json-editor.github.io/json-editor/?schema=N4IgLglmA2CmIC4QAVYCcDOB7AdiANOAJ4AO8SWARgFawDGYBIasAjgK4QsAmiA2iBwBDALbxCQgObiQAMyEA3LGiiwA+nSzRlTaTm7om2ukMi4mZMBhABdQiTRYyaSLGsJQwsYlBhS5EAwwFRxJJgMMOhUSMzwkADEuIIACIX1kgBkhFK94AF8JaR9iMkQQCBwwWGk0cNh5dmhGBAAmAFZCEQqIEXYRRABGAA5OoQAPHr7EAE5pgrlFZVUNLR0PEoCgkLDCWWURUzLNbVrCSBgA+SUVKuTjnUIDBqaygGJZeTaABi+Qeb0DLV1n5SkgthUdiBYDgpggBAc4ExZLAEfAbPNjKYIOZgf4ylRaAwmOdEUgMlgTLELI5nK53KA6FAiMUQZtghC6s9miAAMppZLxNBpRmRLB/QhBUzkXx4sHs0KcoSNbkAYQAguKQIy/JKqizZYF5ZCIlEIDFsXEQAAVAAWEAwyXtyT06Cl3FS7DAWAOkBM0GgRGSskcImSYBtsGSDlgCmx7AdYAA7lggxBYNBuNYziiSNApWVgMBtUQ8gVkoXdbBS0xE6Y6DbisWypjYgA6JsSsD5pAti2tyt/UvzSz0jZlIRoIXM3b7Q5ILuURFnKCklCwKxMdg4CAcWAASSqIncwXYsEIqiP+tBIAJ9EYy4uZVQ95ADic6DpV+lY7l2yY0NhARKXCLBIUoLheEIFhzSXG9w0MOwQCeJUXiQbhQM1XIvzKcEFSHR56hQ5o+Bla90MhLCkAAdSEJpDDydFSyAA=&value=N4IgdghgtgpiBcIQBoQQOZ3gJgKyoDMIA3AewCcBLAFxgH0BjUgGwoRAGICjcAGXlCExgAJjHLsoEZnFSsGEapVJgEwAL6oADjGoBnBAG1Q1AJ472I0ukGRY7AOrTaE9QF11QA==&prompt_before_delete&theme=bootstrap2&iconlib=fontawesome4&object_layout=normal&show_errors=interaction
